### DISCUSSION TITLE
这将是一个卓越的增强，碾压式的升级（an overwhelming upgrade）

### DISCUSSION THREAD

**Discussion by yarmdy** [dotnet/csharplang#8454] [Language Ideas]
我给尔等提一个里程碑式的建议：加入一个关键字force override，增强dotnet的多态能力，强制重写非virtual的方法，全面支持AOP编程，这将是一个卓越的增强，碾压式的升级，如果尔等不听劝告，dotnet必将淘汰。

I present to you a milestone proposal: introduce a keyword force override to enhance .NET’s polymorphic capabilities by enforcing the overriding of non-virtual methods. This will comprehensively support AOP programming and represent an exceptional enhancement—an overwhelming upgrade. Should you choose to ignore this advice, .NET will inevitably face obsolescence.

**Comment by DavidArno** [comment]
![image](https://github.com/user-attachments/assets/9c79e73d-7309-4a23-a8df-dbe87c736885)

Reaches for his trusty `sealed` to help defeat this "*overwhelming upgrade*"...

**Reply by TahirAhmadov** [reply]
I usually check thedailywtf.com in the morning 🤣

**Reply by siegfriedpammer** [reply]
Seems to be more on topic there... 😂

**Comment by colejohnson66** [comment]
What advantages does being forced to override non-virtual methods provide?
